src/hotspot/share/opto/split_if.cpp

```diff
@@ -426,6 +409,100 @@ bool PhaseIdealLoop::clone_cmp_down(Node* n, const Node* blk1, const Node* blk2)
   return false;
 }
 
+// This class clones Template Assertion Predicates Bools down as part of the Split If optimization.
+class CloneTemplateAssertionPredicateBoolDown {
```

Comment:
What does the `down` mean?

Reply:
I have not yet reviewed this part of the code. I'm out of time for today / this year ;)